appsec/appsec.go

```diff
@@ -86,9 +98,43 @@ func setUser(ctx context.Context, id string, userEventType usersec.UserEventType
 // Such events trigger the backend-side events monitoring, such as the Account
 // Take-Over (ATO) monitoring, ultimately blocking the IP address and/or user id
 // associated to them.
+//
+// Deprecated: use [TrackUserLoginSuccess] instead.
 func TrackUserLoginSuccessEvent(ctx context.Context, uid string, md map[string]string, opts ...tracer.UserMonitoringOption) error {
+	login, _, _ := getMetadata(opts)
+	return TrackUserLoginSuccess(ctx, login, uid, md, opts...)
+}
+
+// TrackUserLoginSuccess denotes a successful user login event, which is used
+// by back-end side event monitoring, such as Account Take-Over (ATO)
+// monitoring, ultimately blocking the IP address and/or user ID associated with
+// identified malicious activity.
+//
+// The login is the username that was provided by the user as part of
+// the authentication attempt, and a single user may have multiple different
+// logins (i.e; user name, email address, etc...). The user however has exactly
+// one user ID which canonically identifies them.
+//
+// The provided metata is attached to the successful user login event.
+//
+// This function calso calls [SetUser] with the provided user ID and login, as
+// well as any provided [tracer.UserMonitoringOption]s, and returns an error if
+// the provided user ID is found to be on a configured deny list. See the
+// documentation for [SetUser] for more information.
+func TrackUserLoginSuccess(ctx context.Context, login string, uid string, md map[string]string, opts ...tracer.UserMonitoringOption) error {
+	// We need to make sure the metadata contains the correct information
+	md = maps.Clone(md)
+	if md == nil {
+		md = make(map[string]string, 2)
+	}
+	if uid == "" {
+		log.Warn("appsec: TrackUserLoginSuccess requires a non-empty user ID (uid) in order for user blocking to be effective")
```

Comment:
```suggestion
		log.Error("appsec: TrackUserLoginSuccess requires a non-empty user ID (uid) in order for user blocking to be effective")
```

This one is a real user side error so we don't need to be so conservative about the log level. + using log.Error give us log deduplication since this will surely be in the request hot path, meaning it will flood the logs with it.